union-type.js

```diff
@@ -127,4 +127,22 @@ function Type(desc) {
 
 Type.check = true;
 
+Type.ListOf = function (T) {
+  var List = Type({List:[Array]});
+  var innerType = Type({T: [T]}).T;
+  var validate = List.case({
+    List: function (array) {
+      try{
+        for(var n in array) {
```

Comment:
In addition to the order not being guaranteed, look at this JSBin for another problem that can occur: https://jsbin.com/rixujevexa/1/edit?js,console